homework.md

```diff
@@ -20,48 +20,217 @@ CREATE TABLE stations
 
-``` sql
+As the result of the finding, I fixed the inconsistent date format of time strings
+
+```sql
 UPDATE trips
-SET start_time = ..., end_time = ...
-WHERE ...;
+SET start_time_str = to_char(
+date_trunc('second',
+to_timestamp(start_time_str, 'DD/MM/YYYY HH24:MI:SS')),
+'MM/DD/YYYY HH24:MI')
+WHERE original_filename like '%2017 Q1%' or original_filename like '%2017 Q2%'
 ```
 
-3) Other than the index in class, would we benefit from any other indexes on this table? Why or why not?
+2. Take a look at Postgres's [date functions](https://www.postgresql.org/docs/12/functions-datetime.html), and fill in the missing date data using proper timestamps. You may have to write several queries to do this.
+
+### Question 2 Answer:
+
+With the fix above, I can fill in the missing date data by executing the following sql
+
+```sql
+UPDATE trips
+SET start_time = start_time_str::timestamp
+WHERE start_time_str NOT LIKE '%NULL%'
+
+UPDATE trips
+SET end_time = end_time_str::timestamp
+WHERE end_time_str NOT LIKE '%NULL%'
+```
+
+3. Other than the index in class, would we benefit from any other indexes on this table? Why or why not?
+
+### Question 3 Answer:
+
+Yes, the trips table is a large database, each search sql takes a long time to execute. Adding index for this table makes searching in table faster.
 
 ## Part 3: Data-driven insights
 
 Using the table you made in part 1 and the dates you added in part 2, let's answer some questions about the bike share data
 
-1) Build a mini-report that does a breakdown of number of trips by month
-2) Build a mini-report that does a breakdown of number trips by time of day of their start and end times
-3) What are the most popular stations to bike to in the summer?
-4) What are the most popular stations to bike from in the winter?
-5) Come up with a question that's interesting to you about this data that hasn't been asked and answer it.
+1. Build a mini-report that does a breakdown of number of trips by month
+
+### Question 1 Answer:
+
+Run the following sql
+
+```sql
+SELECT EXTRACT(MONTH FROM start_time) as month, count(*) as number_of_trips
+FROM trips
+GROUP BY month
+```
+
+We get this report
+
+| month | number_of_trips |
+| ----- | :-------------: |
+| 1     |      85155      |
+| 2     |      91350      |
+| 3     |     134172      |
+| 4     |     173791      |
+| 5     |     317373      |
+| 6     |     400394      |
+| 7     |     500019      |
+| 8     |     504696      |
+| 9     |     481362      |
+| 10    |     361550      |
+| 11    |     222996      |
+| 12    |     142465      |
+
+2. Build a mini-report that does a breakdown of number trips by time of day of their start and end times
+3. What are the most popular stations to bike to in the summer?
+
+### Question 2 Answer:
+
+York St / Queens Quay W
+
+```sql
+SELECT from_station_name, count(from_station_name)
+FROM trips
+WHERE EXTRACT(MONTH FROM end_time) BETWEEN 6 AND 8
+GROUP BY from_station_name
+ORDER BY count DESC
+LIMIT 1
+```
```

Comment:
Confused, is this answering Question 3? If so should it not be `to_station_name`?